### ISSUE TITLE
Tests fail on latest macOS / Safari

### ISSUE THREAD

**Issue by ercgrat** [jhildenbiddle/canvas-size#10]
This is a question, assuming it's my lack of understanding and not a bug, really not sure.

I have a completed t-shirt designer application that relies on drawing a large canvas in memory to render a design created by the user. When I encountered issues on mobile devices, I turned to this repo as a solution to warn the user that the tool is unavailable on their device. Thanks!

Unexpectedly, my fix to disable my site on mobile using the canvas test also broke Safari for desktop Macs. The result of calling `test({ width: 4189, height: 4790.8 })` is false, but when I query for the `maxArea()`, the result I get on success has width and height `8192`. Why would the test be failing even though the dimensions are within the bounds?

To unblock my progress I am switching to a solution that checks the result of `maxArea()` but I am concerned that this might not reflect what will happen when I try to render the design.

**Comment by ercgrat** [comment]
Possibly related, using the promise version of maxArea returns 1/1 instead of 8192/8192.

**Comment by jhildenbiddle** [comment]
Hi @ercgrat.

That strange. I wish I had an easy answer for you, but unfortunately I'm not seeing the same behavior when I test locally. Both of the following examples successfully render a canvas element at the specified resolution using Safari 14.0.3 on macOS 11.2.1:

```js
canvasSize.test({
  width: 4189, 
  height: 4790.8,
  onError(width, height, benchmark) {
    console.log('FAIL', width, height, benchmark);
  },
  onSuccess(width, height, benchmark) {
    console.log('PASS', width, height, benchmark);
  }
});

// PASS 4189 4790.8 26
```

```js
canvasSize.test({
  width: 4189, 
  height: 4790.8,
  usePromise: true
})
.then(({width, height, benchmark}) => {
  console.log('PASS', width, height, benchmark);
})
.catch(({width, height, benchmark}) => {
  console.log('FAIL', width, height, benchmark);
});

// PASS 4189 4790.8 26
```

My first suggestion is to try and replicate the issue with a simplified demo using https://codesandbox.io or a similar service. If you are using a version of Safari other than 14.0.3, consider downloading the [Safari Technology Preview](https://developer.apple.com/safari/technology-preview/) to see if this is an issue specific to the version of Safari you are using.

Not sure if that helps, but perhaps it will get the ball rolling.

**Comment by ercgrat** [comment]
On the latest Mac OS (Big Sur) and latest Safari (14) on BrowserStack, using the sandbox linked below, I get:

`FAIL 4189 4790.8 15`
`FAIL 4189 4790.8 204`
https://codesandbox.io/s/elegant-kalam-y0b79?file=/src/App.js

**Comment by jhildenbiddle** [comment]
Thanks for creating the codesandbox demo, @ercgrat. Very helpful.

It appears that the issue is Safari's handling of floating point values when rending `canvas` elements. Changing the float value above (4790.8) to an integer (4790) fixes the issue in your demo and in my automated tests (locally and on BrowserStack). As you mentioned, Chrome and Firefox handle the same floating point value without issue (presumably because they are rounding floats under the hood when working with `canvas` elements).

The fix is now available on the `dev` [branch](https://github.com/jhildenbiddle/canvas-size/tree/dev). You can test this branch in your app to verify that it addresses your issue:

```shell
npm install git://github.com/jhildenbiddle/canvas-size.git#dev --save
```

Once confirmed that the fix works as expected in your app, I'll push a patch release to npm.

Thanks again for taking the time to file the PR, @ercgrat. Much appreciated.

**Comment by ercgrat** [comment]
Wow, how obscure. Thanks for addressing so quickly! Will verify ASAP and comment back here.

**Comment by ercgrat** [comment]
@jhildenbiddle I took a look at your branch, and I think `Math.ceil()` would be more appropriate than `Math.round()` on the off chance that the number I am testing is greater than the canvas dimension limit by a fraction. For example, if I wanted to render a canvas with with 1000.4 and the browser limit was 1000, your test would say it was valid. The only reason I am passing a float is because I am calculating a centimeters to pixel conversion.

**Comment by jhildenbiddle** [comment]
@ercgrat --

Yep. That's the right approach. Good catch. I'll make the change later this morning and reply here when it's ready.

Thanks!

**Comment by jhildenbiddle** [comment]
All set on [`dev`](https://github.com/jhildenbiddle/canvas-size/tree/dev), @ercgrat. Give it a shot and let me know how it works out. Assuming all is well, I'll get a patch release ASAP.

**Comment by jhildenbiddle** [comment]
Fixed in v1.2.3.

Thanks again, @ercgrat. :)
